### DISCUSSION TITLE
v1.14.0

### DISCUSSION THREAD

**Discussion by yuanyao-nv** [onnx/onnx#5203] [Release]
ONNX v1.14.0 is now available with exciting new features! We would like to thank everyone who contributed to this release! Please visit [onnx.ai](https://onnx.ai/) to learn more about ONNX and associated projects.

# New operators
[DeformConv](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#deformconv-19) added in https://github.com/onnx/onnx/pull/4783

# Operator extensions
[Equal](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#equal-19) - Support for string data type added in https://github.com/onnx/onnx/pull/4828
[AveragePool](url) - New attribute ``dilations`` https://github.com/onnx/onnx/pull/4790
[Pad](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#pad-19) - Added new ``wrap`` to the ``mode`` attribute to support circular padding https://github.com/onnx/onnx/pull/4793
[Resize](https://github.com/onnx/onnx/pull/4862) - Added ``half_pixel_symmetric`` to the ``coordinate_transformation_mode`` attribute https://github.com/onnx/onnx/pull/4862

# IR updates (bump to 9)
- Support attributes with default values: https://github.com/onnx/onnx/pull/4911
- Added 4 new 8-bit floating point data types: https://github.com/onnx/onnx/pull/4805

# Backend tests
Replaced real models with light models in backend tests. https://github.com/onnx/onnx/pull/4861 https://github.com/onnx/onnx/pull/4960

# Support Protobuf v21
Now ONNX supports Protobuf v21: https://github.com/onnx/onnx/pull/4956

# Deprecation notice
- Python 3.7 support will be deprecated due to EOL in next release: https://github.com/onnx/onnx/pull/5191
- onnx-weekly package will be deprecated in TestPyPI. Please use them from PyPI instead: https://github.com/onnx/onnx/issues/4930
- Properties in FormalParameter will be deprecated in future release. Please use newer properties name:  https://github.com/onnx/onnx/issues/5074
- Variables from mapping.py will be deprecated and become private implementation details. Please use public functions to get corresponding types from helper.py instead: https://github.com/onnx/onnx/issues/4554

# Installation Notice
You can upgrade to the latest release using `pip install onnx --upgrade` or build from source following the README [instructions](https://github.com/onnx/onnx/tree/rel-1.14.0#build-onnx-from-source).

# Contributors
Thanks to these individuals for their contributions in this release since last 1.13.0 release: @jcwchen, @andife, @gramalingam, @xadupre, @justinchuby, @liqunfu, @yuanyao-nv, @jbachurski, @p-wysocki, @prasanthpul, @jantonguirao, @take-cheeze, @smk2007, @AlexandreEichenberger, @snnn, @daquexian, @linkerzhang. 

<hr /><em>This discussion was created from the release <a href='https://github.com/onnx/onnx/releases/tag/v1.14.0'>v1.14.0</a>.</em>

**Comment by take-cheeze** [comment]
Thank you for the release!
IR version update should be noted since it may break some code relying on `onnx.helper.make_model`

And after https://github.com/onnx/onnx/pull/4911 the IR version docs isn't updated
https://github.com/onnx/onnx/blame/v1.14.0/onnx/onnx.proto#L106

**Reply by jcwchen** [reply]
Hi @take-cheeze,
Good point. Thank you for the suggestion! The release note has been update accordingly with explicit IR bump notice  and the PR for updating TBD date is here: https://github.com/onnx/onnx/pull/5208.

**Reply by cbourjau** [reply]
Is there a safe way to generate models using `1.14.0` with the previous (or any explicit) IR version?  From the point of view of somebody maintaining a model-producing library an implicit bump of the IR may be quite disruptive. It will take a while for runtimes to catch up and be package for various distribution channels.

**Reply by jcwchen** [reply]
@cbourjau You are right. onnx.helper can be disruptive sometimes, because currently `onnx.helper.make_model` will [always use the latest IR_VERSION](https://github.com/onnx/onnx/blob/f8da6749be56401880241221d01ffeb04c9cac95/onnx/helper.py#L287) and the latest opset_version by default, although it can specify the opset_version (and also ir_version). Other than that, `onnx.helper.make_model_gen_version` can generate models with specified opset_version and corresponding IR_VERSION, which is typically a safer way to get a stable model.

~Now users can only manually fix the IR_VERSION after `onnx.helper.make_model`. However, it seems reasonable to me that onnx should enable helper.make_model to get specified IR_VERSION.~

**Reply by gramalingam** [reply]
A correction: my reading of [this line](https://github.com/onnx/onnx/blob/f8da6749be56401880241221d01ffeb04c9cac95/onnx/helper.py#L306) is that if the caller explicitly specifies `ir_version` via `ir_version=...` in the call that takes precedence in `make_model` (both versions). Only if the caller does not specify ir_version, is `make_model` going to have a release-dependent behavior.

**Reply by jcwchen** [reply]
Thank you @gramalingam for the correction! Sorry I misread that. Yes, current `onnx.helper.make_model` is able to created a model with specified ir_version by `onnx.helper.make_model(graph, ir_version=8)`. I have revised my previous comment to prevent confusion.

**Reply by take-cheeze** [reply]
Thank you for updating note!
I hope onnxruntime supporting new IR would be released soon.

**Reply by cbourjau** [reply]
Thanks for the clarification!
